File: functions/test_list_plans.js
const { initializeApp } = require('firebase-admin/app');
const { getFirestore } = require('firebase-admin/firestore');

const app = initializeApp({ projectId: 'planner-cac2a' });
const db = getFirestore();

async function test() {
  const users = await db.collection('users').limit(1).get();
  if (users.empty) {
    console.log("No users found");
    return;
  }
  const uid = users.docs[0].id;
  const plans = await db.collection(`users/${uid}/plans`).get();
  for (const plan of plans.docs) {
    const details = await plan.ref.collection('details').doc('main').get();
    console.log(plan.id, "details exists:", details.exists, "details data:", details.data());
  }
}
test().catch(console.error);

File: functions/test_list_plans.py
import firebase_admin
from firebase_admin import credentials
from firebase_admin import firestore

cred = credentials.ApplicationDefault()
firebase_admin.initialize_app(cred, {
  'projectId': 'planner-cac2a',
})

db = firestore.client()
users = db.collection('users').limit(1).get()
if not users:
    print("No users")
    exit()
uid = users[0].id
plans = db.collection(f'users/{uid}/plans').get()
for plan in plans:
    details_doc = db.collection(f'users/{uid}/plans/{plan.id}/details').document('main').get()
    print(plan.id, "details exists:", details_doc.exists, "data:", details_doc.to_dict())
